File: Python/1614.py
# 1614. Maximum Nesting Depth of the Parentheses
# https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/

class Solution:
  def maxDepth(self, s: str) -> int:
    cnt = 0
    out = 0

    for c in s:
      match c:
        case '(':
          cnt += 1

          if (cnt > out):
            out = cnt

        case ')':
          cnt -= 1

        # case _:
          # do nothing

    return out

File: Rust/1614.rs
/**
 * 1614. Maximum Nesting Depth of the Parentheses
 * https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/
 */

impl Solution {
  pub fn max_depth(s: String) -> i32 {
    let mut cnt: i32 = 0;
    let mut out: i32 = 0;

    for c in s.chars() {
      match c {
        '(' => {
          cnt += 1;

          if cnt > out {
            out = cnt;
          }
        }

        ')' => {
          cnt -= 1;
        }

        _ => {}
      }
    }

    return out;
  }
}
